File: liqp-core/src/main/java/liqp/tags/Unless.java
package liqp.tags;

import liqp.tag.LTag;
import liqp.node.LNode;
import liqp.context.LContext;

public class Unless extends LTag {

    /*
     * Mirror of if statement
     */
    @Override
    public Object render(LContext context, LNode... nodes) {

        for (int i = 0; i < nodes.length - 1; i += 2) {

            Object exprNodeValue = nodes[i].render(context);
            LNode blockNode = nodes[i + 1];

            if (context.isFalse(exprNodeValue)) {
                return blockNode.render(context);
            }
        }

        return "";
    }
}

File: liqp-core/src/main/kotlin/liqp/tags/Unless.kt
package liqp.tags

import liqp.ControlResult
import liqp.context.LContext
import liqp.node.LNode
import liqp.tag.LTag

class Unless : LTag() {

  /**
   * Mirror of if statement
   */
  override fun render(context: LContext, vararg nodes: LNode): Any? {

    val iterator = nodes.iterator()
    while (iterator.hasNext()) {
      val exprNodeValue = iterator.next().render(context)
      val blockNode = iterator.next()

      if (context.isFalse(exprNodeValue)) {
        return blockNode.render(context)
      }
    }

    return ControlResult.NO_CONTENT
  }
}
